### PR TITLE
Enable ability to open project through a url (issue 124)

### PR DESCRIPTION
**Pull request by microsoftsam**
fixes issue #124

### CODE REVIEW

File: webapp/src/app.tsx [hunk 1]
```diff
@@ -143,6 +144,21 @@ class ScriptSearch extends data.Component<ISettingsProps, ScriptSearchState> {
         return this.prevData
     }
 
+    fetchUrlData(): Cloud.JsonScript[] {
+        if (this.state.packages) return []
+        
+        if (this.state.searchFor) {
+            let m = /^((https:\/\/)?codethemicrobit.com\/)?(api\/oembed\?url=.*%2F([^&]*)&.*?|(.+))$/i.exec(this.state.searchFor.trim());
```
Comment (pelikhan) [inline webapp/src/app.tsx]: use pxt.appTarget.appTheme.embedUrl to determine the domain name

File: webapp/src/app.tsx [hunk 2]
```diff
@@ -197,6 +214,14 @@ class ScriptSearch extends data.Component<ISettingsProps, ScriptSearchState> {
                     .done()
             }
         }
+        const installScript = (scr: Cloud.JsonScript) => {
+            if (this.modal) this.modal.hide();
+            if (!this.state.packages) {
+                workspace.installByIdAsync(scr.id)
```
Comment (pelikhan) [inline webapp/src/app.tsx]: does this pop a gear wheel dialog? If it is loaded from the web, it might take a few seconds.

File: pxtlib/github.ts [hunk 3]
```diff
@@ -144,7 +144,7 @@ namespace pxt.github {
     }
 
     export function parseRepoUrl(url: string) {
-        let m = /^((https:\/\/)?github.com\/)?([^/]+\/[^/]+)(#(\w+))?$/i.exec(url.trim());
+        let m = /^((https:\/\/)?github.com\/)?([^/]+\/[^/#]+)(#(\w+))?$/i.exec(url.trim());
```
Comment (pelikhan) [inline pxtlib/github.ts]: Unrelated fix?

File: webapp/src/app.tsx (hunk 2 above)
Comment (microsoftsam) [inline webapp/src/app.tsx]: Fixed.

File: webapp/src/app.tsx (hunk 1 above)
Comment (microsoftsam) [inline webapp/src/app.tsx]: Fixed.

File: pxtlib/github.ts (hunk 3 above)
Comment (microsoftsam) [inline pxtlib/github.ts]: Yes
